merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetProfileSegmentsAction.java

```diff
@@ -67,12 +67,14 @@ public <Dynamics> List<Pair<Duration, Dynamics>> get(
   }
 
   private <Dynamics> Dynamics parseDynamics(final Reader jsonStream, final JsonParser<Dynamics> dynamicsP) {
-    final var json = Json.createReader(jsonStream).readValue();
-    return dynamicsP
-        .parse(json)
-        .getSuccessOrThrow(
-            failureReason -> new Error(
-                "Corrupt profile dynamics: " + failureReason.reason()));
+    try(final var reader = Json.createReader(jsonStream)) {
```

Comment:
FWIW, there are a few other places in 07b414e35fae1d0baf2edd44ff17240adfe7f3ec that my earlier comment on closing a received reader applies to, and they would all benefit from the new `getJsonColumn`. I could comment on each if you like, but they're all visible at that commit.

Reply:
True. Thank you 🙏